features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java

```diff
@@ -181,71 +190,66 @@ private void provision(final OnmsMinion minion,
-            requisitionInterface.putMonitoredService(requisitionMonitoredService);
-
-            requisitionNode = new RequisitionNode();
-            requisitionNode.setNodeLabel(minion.getId());
-            requisitionNode.setForeignId(minion.getLabel() != null
-                                         ? minion.getLabel()
-                                         : minion.getId());
-            requisitionNode.setLocation(minion.getLocation());
-            requisitionNode.putInterface(requisitionInterface);
-
-            nextRequisition.putNode(requisitionNode);
-            nextRequisition.setDate(new Date());
-            deployedForeignSourceRepository.save(nextRequisition);
-            deployedForeignSourceRepository.flush();
-
-            alteredForeignSources.add(nextForeignSource);
-        }
+            RequisitionNodeEntity requisitionNode = nextRequisition.getNode(minion.getId());
+            if (requisitionNode == null) {
+                final RequisitionMonitoredServiceEntity requisitionMonitoredService = new RequisitionMonitoredServiceEntity();
+                requisitionMonitoredService.setServiceName("Minion-Heartbeat");
 
-        for (final String alteredForeignSource : alteredForeignSources) {
-            final EventBuilder eventBuilder = new EventBuilder(EventConstants.RELOAD_IMPORT_UEI, "Web");
-            eventBuilder.addParam(EventConstants.PARM_URL, String.valueOf(deployedForeignSourceRepository.getRequisitionURL(alteredForeignSource)));
+                final RequisitionInterfaceEntity requisitionInterface = new RequisitionInterfaceEntity();
+                requisitionInterface.setIpAddress("127.0.0.1");
+                requisitionInterface.addMonitoredService(requisitionMonitoredService);
 
-            try {
-                eventProxy.send(eventBuilder.getEvent());
-            } catch (final EventProxyException e) {
-                throw new DataAccessResourceFailureException("Unable to send event to import group " + alteredForeignSource, e);
+                requisitionNode = new RequisitionNodeEntity();
+                requisitionNode.setNodeLabel(minion.getId());
+                requisitionNode.setForeignId(minion.getLabel() != null
+                        ? minion.getLabel()
+                        : minion.getId());
+                requisitionNode.setLocation(minion.getLocation());
+                requisitionNode.addInterface(requisitionInterface);
+
+                nextRequisition.addNode(requisitionNode);
+                nextRequisition.setLastUpdate(new Date());
+                requisitionService.saveOrUpdateRequisition(nextRequisition);
+
+                alteredForeignSources.add(nextForeignSource);
             }
+            return null;
+        });
+
+        for (final String alteredForeignSource : alteredForeignSources) {
+            requisitionService.triggerImport(
+                    new ImportRequest("Web").withForeignSource(alteredForeignSource));
```

Comment:
Originally all Import Requests had `Web` as their source, no matter their origin. This I probably just copied and forgot to adjust. Changed it to `this`.